### ISSUE TITLE
Some Confusions About the Config

### ISSUE THREAD

**Issue by dalek-who** [pettingllms-ai/PettingLLMs#13]
I have read the configuration documentation in detail, but I have some confusion about a few fields:

1.  `specialization`: "prompt" for prompt optimization. Does this refer to manual optimization rather than automatic? In other words, if set to "prompt", does it only apply to **evaluate** and not to **train**?
2.  `env.dataset` and `env.benchmark`: Can they be considered as `train_data` and `val_data` respectively? Are multiple `val_data` supported?
3.  `base_models.policy_X`: If N agents share the same base model (i.e., when `specialization` is `lora` or `prompt`), is it sufficient to define only one `policy_0`, or do we need to define N `policy_X` with the same path but different names?
4.  `agent_policy_configs.num_agents`, `agent_policy_configs.policy_list`, and `agent_policy_configs.agent_configs.agent_X`. Since all agents are already defined in `agent_policy_configs.agent_configs.agent_X`, does that mean `agent_policy_configs.num_agents` and `agent_policy_configs.policy_list` are redundant and should be removed?
5.  `agent_policy_configs.policy_list`: Its actual meaning seems to be "agent_list", as it is a list of `agent_policy_configs.agent_configs.agent_X.name`. The name `policy_list` is somewhat confusing.
6.  `env.max_turns`, `agent_policy_configs.num_agents`, and `multi_agent_interaction.num_interacting_agents`: I hope the relationship between these three can be clarified.
7.  `training.train_batch_size`, `training.total_training_steps`, and `training.epoch_size`: Does this mean EnvBatch selects `train_batch_size` samples for rollout each time, the model parameters are updated for a total of `total_training_steps` during training, with `epoch_size` steps per epoch? Should there be a certain relationship between the values of `train_batch_size` and `epoch_size`? Is the batch_size the same as `train_batch_size` during evaluation?
8.  `trainer.resample_freq`: Does it mean the training data is shuffled every `resample_freq` steps? Will the training data maintain the original order from the dataset?
9.  When defining an agent, how should custom kwargs be passed in?
10. Is it possible to specify which directory to output the state from each evaluation during training to?
11. `models.model_X`: Should the number of these match the number of `base_models.policy_X`?
12. How to config **not to train** some agents in a multi-agent system? 

I hope you can clarify the above questions. I believe other users might have similar confusions.

**Comment by dalek-who** [comment]
I find answer for some questions:
> 1.specialization: "prompt" for prompt optimization. Does this refer to manual optimization rather than automatic? In other words, if set to "prompt", does it only apply to evaluate and not to train?

`specialization: "prompt"` can be set for training the shared base model with different prompts.

> 3.base_models.policy_X: If N agents share the same base model (i.e., when specialization is lora or prompt), is it sufficient to define only one policy_0, or do we need to define N policy_X with the same path but different names?

According to https://github.com/pettingllms-ai/PettingLLMs/blob/main/pettingllms/config/math/math_L2_lora.yaml , only define policy_0 is enough.

> 11.models.model_X: Should the number of these match the number of base_models.policy_X?

True
